editor/import/resource_importer_wav.cpp

```diff
@@ -95,208 +108,55 @@ void ResourceImporterWAV::get_import_options(const String &p_path, List<ImportOp
-				ERR_FAIL_V_MSG(ERR_INVALID_DATA, "Invalid amount of bits in the sample (should be one of 8, 16, 24 or 32).");
-			}
-
-			if (compression_code == 3 && format_bits % 32) {
-				ERR_FAIL_V_MSG(ERR_INVALID_DATA, "Invalid amount of bits in the IEEE float sample (should be 32 or 64).");
-			}
-
-			/* Don't need anything else, continue */
-			format_found = true;
-		}
-
-		if (chunkID[0] == 'd' && chunkID[1] == 'a' && chunkID[2] == 't' && chunkID[3] == 'a' && !data_found) {
-			/* IS DATA CHUNK */
-			data_found = true;
-
-			if (!format_found) {
-				ERR_PRINT("'data' chunk before 'format' chunk found.");
-				break;
-			}
-
-			frames = chunksize;
-
-			if (format_channels == 0) {
-				ERR_FAIL_COND_V(format_channels == 0, ERR_INVALID_DATA);
-			}
-			frames /= format_channels;
-			frames /= (format_bits >> 3);
-
-			/*print_line("chunksize: "+itos(chunksize));
-			print_line("channels: "+itos(format_channels));
-			print_line("bits: "+itos(format_bits));
-			*/
-
-			data.resize(frames * format_channels);
-
-			if (compression_code == 1) {
-				if (format_bits == 8) {
-					for (int i = 0; i < frames * format_channels; i++) {
-						// 8 bit samples are UNSIGNED
-
-						data.write[i] = int8_t(file->get_8() - 128) / 128.f;
-					}
-				} else if (format_bits == 16) {
-					for (int i = 0; i < frames * format_channels; i++) {
-						//16 bit SIGNED
-
-						data.write[i] = int16_t(file->get_16()) / 32768.f;
-					}
-				} else {
-					for (int i = 0; i < frames * format_channels; i++) {
-						//16+ bits samples are SIGNED
-						// if sample is > 16 bits, just read extra bytes
-
-						uint32_t s = 0;
-						for (int b = 0; b < (format_bits >> 3); b++) {
-							s |= ((uint32_t)file->get_8()) << (b * 8);
-						}
-						s <<= (32 - format_bits);
-
-						data.write[i] = (int32_t(s) >> 16) / 32768.f;
-					}
-				}
-			} else if (compression_code == 3) {
-				if (format_bits == 32) {
-					for (int i = 0; i < frames * format_channels; i++) {
-						//32 bit IEEE Float
-
-						data.write[i] = file->get_float();
-					}
-				} else if (format_bits == 64) {
-					for (int i = 0; i < frames * format_channels; i++) {
-						//64 bit IEEE Float
-
-						data.write[i] = file->get_double();
-					}
+	AudioStreamWAV::LoopMode loop_mode = AudioStreamWAV::LOOP_DISABLED;
+	if (import_loop_mode == 0) {
+		for (uint32_t meta = 0; meta < wav.metadataCount; meta++) {
+			drwav_metadata md = wav.pMetadata[meta];
+			if (md.type == drwav_metadata_type_smpl) {
+				if (md.data.smpl.sampleLoopCount) {
+					drwav_smpl_loop loop = md.data.smpl.pLoops[0];
+					if (loop.type == drwav_smpl_loop_type_forward)
+						loop_mode = AudioStreamWAV::LOOP_FORWARD;
+					else if (loop.type == drwav_smpl_loop_type_pingpong)
+						loop_mode = AudioStreamWAV::LOOP_PINGPONG;
+					else if (loop.type == drwav_smpl_loop_type_backward)
+						loop_mode = AudioStreamWAV::LOOP_BACKWARD;
+					loop_begin = loop.firstSampleByteOffset;
+					loop_end = loop.lastSampleByteOffset;
```

Comment:
The "Byte" in the name here is misleading, it's actually the frame position of loop begin/end.